File: RecipeApp/app/src/main/java/skydu/android/recipeapp/database/AppDatabaseHelper.java
package skydu.android.recipeapp.database;

import android.content.Context;

import androidx.room.Room;

public class AppDatabaseHelper {
    private static AppDatabase instance;

    public static synchronized AppDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, DatabaseConstant.DB_NAME)
                    .build();
        }
        return instance;
    }
}

File: RecipeApp/app/src/main/java/skydu/android/recipeapp/database/AppDatabaseHelper.kt
package skydu.android.recipeapp.database

import android.content.Context
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.sqlite.db.SupportSQLiteDatabase
import skydu.android.recipeapp.thread.AppExecutors

object AppDatabaseHelper {
    private var instance: AppDatabase? = null
    @Synchronized
    fun getInstance(context: Context): AppDatabase {
        if (instance == null) {
            instance = Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                DatabaseConstant.DB_NAME
            )
                .addCallback(object : RoomDatabase.Callback() {
                    override fun onCreate(db: SupportSQLiteDatabase) {
                        super.onCreate(db)
                        AppExecutors.getInstance().diskIO().execute {
                            getInstance(context).recipeDao().run {
                                for (i in 0..10) {
                                    this.insert(
                                        RecipeEntity(
                                            0,
                                            "Nama $i",
                                            false,
                                            false,
                                            "bahan $i",
                                            "cara $i"
                                        )
                                    )
                                }
                            }
                        }
                    }
                })
                .build()
        }
        return instance!!
    }
}